vip-jetpack/connection-pilot/class-jetpack-connection-controls.php

```diff
@@ -0,0 +1,278 @@
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		if ( ! $body ) {
+			return new WP_Error( 'jp-cxn-pilot-empty-body', 'Failed to test connection (empty response body).' );
+		}
+
+		$result       = json_decode( $body );
+		$is_connected = isset( $result->connected ) ? (bool) $result->connected : false;
+		if ( ! $is_connected ) {
+			return new WP_Error( 'jp-cxn-pilot-not-connected', 'Connection test failed (WP.com does not think this site is connected or there are authentication or other issues).' );
+		}
+
+		return true;
+	}
+
+	/**
+	 * (Re)connect a site to Jetpack.
+	 *
+	 * Creates the VIP user if needed, provisions the plan with WP.com, and re-runs the connection checks to ensure it all worked.
+	 *
+	 * @param bool $skip_connection_tests Skip if we've already run the checks before this point.
+	 * @param bool $disconnect Set to true if it should disconnect Jetpack at the start.
+	 *
+	 * @return mixed bool|WP_Error True if JP was (re)connected, WP_Error otherwise.
+	 */
+	public static function connect_site( $skip_connection_tests = false, $disconnect = false ) {
+		if ( ! self::validate_constants() ) {
+			return new WP_Error( 'jp-cxn-pilot-missing-constants', 'This is not a valid VIP Go environment or some constants are missing.' );
+		}
+
+		if ( ! $skip_connection_tests && ! $disconnect ) {
+			$connection_test = self::jetpack_is_connected();
+
+			if ( true === $connection_test ) {
+				// Abort since the site is already connected to JP, and we aren't okay with disconnecting.
+				return new WP_Error( 'jp-cxn-pilot-already-connected', 'Jetpack is already properly connected.' );
+			}
+		}
+
+		if ( $disconnect ) {
+			Jetpack::disconnect();
+		}
+
+		$user = self::maybe_create_user();
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		// Put the machine user in charge of things now.
+		wp_set_current_user( $user->ID );
+
+		// Register/connect the site to WP.com.
+		$provision_result = self::provision_site( $user->ID );
+
+		if ( is_wp_error( $provision_result ) ) {
+			return $provision_result;
+		}
+
+		// Without this, Jetpack can incorrectly think it's still disconnected.
+		self::refresh_options_cache();
+
+		// Run the tests again and return the result 🤞.
+		return self::jetpack_is_connected();
+	}
+
+	/**
+	 * Provision the site with WP.com.
+	 *
+	 * @see https://github.com/Automattic/host-partner-documentation/blob/master/jetpack/plan-provisioning.md
+	 *
+	 * @param int $user_id The VIP machine user ID.
+	 *
+	 * @return mixed bool|WP_Error True if provisioning worked, WP_Error otherwise.
+	 */
+	private static function provision_site( $user_id ) {
+		// TODO: This is a high-risk possible problem on the CLI/CRON containers. Needs testing.
+		$script_path = __DIR__ . '/bin/partner-provision.sh';
+
+		// Can use JP core's script once the changes here are made: https://github.com/Automattic/vip-go-mu-plugins/pull/986.
```

Comment:
Rather than include a 3rd, slightly different copy of this script, can we use the one that currently works, `jetpack-start/bin/partner-provision.sh`, then update this as part of #986?

Reply:
For sure. Need to make the two adjustments - the one in the above issue, as well as changing the `partner_tracking_id` arg (I've checked in with the JP team about this and got the go-ahead).

Reply:
@WPprodigy I'm trying to sort this out - our `jetpack-start` already has a `partner-provision.sh` script (which is different than the core JP one)...this is what we currently use, is there any issue using that one instead of this version?

The first issue only applies to using the core provision script (which we're not currently)...and it's not clear what the `partner_tracking_id` one is referring to.

Reply:
I think I answered my question - the script in this PR returns JSON to be parsed by PHP, where our existing one from JP start does not, doh.